### PR TITLE
Cache unittest

### PR DESCRIPTION
**Pull request by dan-trang**

### CODE REVIEW

File: olaf/common/oresat_file_cache.py [hunk 1]
```diff
@@ -24,12 +23,15 @@ def __init__(self, dir_path: str):
         self._data = []
         self._lock = Lock()
 
-        Path(self._dir).mkdir(parents=True, exist_ok=True)
+        if(isfile(abspath(self._dir))):
+            print("Cannot create new directory with an existing file name.")
```
Comment (ryanpdx) [inline olaf/common/oresat_file_cache.py]: Don't use print, raise an exception

Comment (ryanpdx) [reply olaf/common/oresat_file_cache.py]: Also, the else is not needed if the exception is raised.

File: olaf/common/oresat_file_cache.py [hunk 2]
```diff
@@ -85,10 +87,9 @@ def remove(self, file_name: str):
         with self._lock:
             for f in self._data:
                 if f.name == file_name:
-                    remove(f.path)
-                    self._data.remove(f)
```
Comment (ryanpdx) [inline olaf/common/oresat_file_cache.py:89]: This line is still needed, it remove the item that is getting deleted from the list

File: tests/common/test_ecss.py [hunk 3]
```diff
@@ -1,4 +1,5 @@
 import unittest
+
```
Comment (ryanpdx) [inline tests/common/test_ecss.py]: This new line is not needed

File: tests/common/test_oresat_file_cache.py [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+
+from shutil import rmtree
+
+from os import rmdir, remove, listdir
+
+from os.path import basename, abspath, exists
+
+from olaf.common.oresat_file_cache import OreSatFileCache
+
+from olaf.common.oresat_file import new_oresat_file
```
Comment (ryanpdx) [inline tests/common/test_oresat_file_cache.py]: Don't need spaces between all imports.

Import style used in OLAF is:
python core imports group
other library imports group
internal (OLAF) imports group

Should have a new line between each group of imports.

File: tests/common/test_oresat_file_cache.py [hunk 5]
```diff
@@ -0,0 +1,299 @@
+import sys
+sys.path.append('../../')
```
Comment (ryanpdx) [inline tests/common/test_oresat_file_cache.py]: Not needed.

File: tests/common/test_oresat_file_cache.py [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+        self.assertEqual(result.dir, abspath(dir_name))
+        self.assertFalse(result._data)
+        self.assertTrue(result._lock)
+        self.assertEqual(OreSatFileCache.__len__(result), 0)
+        rmdir(abspath(dir_name))
+
+        # TEST: existing file name used as dir name
+        dir_name = 'existing_file.xyz'
+        with open(dir_name, 'w'):
+            pass
```
Comment (ryanpdx) [inline tests/common/test_oresat_file_cache.py:47]: Could use `touch()` from the `pathlib` library instead

File: tests/common/test_oresat_file_cache.py [hunk 7]
```diff
@@ -0,0 +1,299 @@
...
+        self.assertEqual(result.dir, abspath(dir_name))
+        self.assertEqual(len(listdir(result._dir)), 0)
+
+        # cleanup
+        rmdir(dir_name)
+        remove(abspath(file_name))
+
+
+if __name__ == '__main__':
+    unittest.main()
```
Comment (ryanpdx) [inline tests/common/test_oresat_file_cache.py]: Not needed

File: tests/common/test_oresat_file_cache.py [hunk 8]
```diff
@@ -0,0 +1,299 @@
+import sys
+sys.path.append('../../')
+
+import unittest
+
+import threading
+
+from time import sleep
```
Comment (ryanpdx) [inline tests/common/test_oresat_file_cache.py]: Not used

**Review by ryanpdx — CHANGES_REQUESTED** [review]
Looks good! 
Most comments are style changes,

**Comment by dan-trang** [comment]
Cleaned up the requested changes!

File: tests/common/test_oresat_file_cache.py (hunk 6 above)
Comment (dan-trang) [reply tests/common/test_oresat_file_cache.py:47]: Decided to stick to using the open() function instead of importing pathlib to use touch().

**Review by ryanpdx — APPROVED** [review]

**Pull request merged**